test_runner/regress/test_wal_restore.py

```diff
@@ -38,7 +38,7 @@ def test_wal_restore(
     pg_distrib_dir: Path,
 ):
     env = neon_env_builder.init_start()
-    env.neon_cli.create_branch("test_wal_restore")
+    env.create_branch("test_wal_restore")
```

Comment:
Do you mean "... rather than `neon_simple_env`"? Probably just a matter of which existing test it was copy-pasted from when the test was created.

I'd like to see us use `neon_simple_env` whenever you don't need the extra flexibility of `neon_env_builder`, because it's a good quick hint that the test doesn't do anything funny with the storage. And is a candidate for sharing the storage environment with other tests in the future.